### PR TITLE
Optimize Auditing DB Schema

### LINKED ISSUE #2457: Auditing: Polish DB Schema
Add:

- DB indexes: `actionType`, `operationType`, `correlation_id`, `created_time`
- Constraints: `auditOperation `, `auditAction ` (e.g. not-null)
- add a column: `parent_correlation_id` (nullable)

to relevant schema fields

### LINKED ISSUE #2497: Change audit schema table names to use ALL_CAPS
Use underscore and all capital letters for table names to be consistent with the rest of the SCDF schema.

### PR DESCRIPTION
**Pull request by tzolov**
- For MySQL, PostgreSQL and MSSQL add indexes for the audit_Action, audit_Operation, correlation_id, created_On columns for the AUDIT_RECORDS table
 - For MySQL, PostgreSQL and MSSQL add not-null constraints on audit_Action and audit_Operations columns for the AUDIT_RECORDS table

Resolves #2457
Resolves #2497

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/resources/schemas/mysql/jpa.sql [hunk 1]
```diff
@@ -13,16 +13,21 @@
 
     create table AUDIT_RECORDS (
        id bigint not null,
-        audit_Action bigint,
+        audit_Action bigint NOT NULL,
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-server-core/src/main/resources/schemas/mysql/jpa.sql]: Not related to the scope of this PR but just curious to know if there is an implicit meaning of using uppercase letter at the second word on some of these column names (ex: audit_Action, audit_Operation etc.,)?

**Comment by ilayaperumalg** [comment]
LGTM

File: spring-cloud-dataflow-server-core/src/main/resources/schemas/mysql/jpa.sql (hunk 1 above)
Comment (markpollack) [reply spring-cloud-dataflow-server-core/src/main/resources/schemas/mysql/jpa.sql]: It is a bit odd...

**Comment by ilayaperumalg** [comment]
Squashed, rebased and merged as 960c8599

**Pull request closed without merging**